src/test/java/liuni/LiUniResourceTests.java

```diff
@@ -0,0 +1,163 @@
+import static org.mockito.Mockito.when;
+
+public class LiUniResourceTests {
+    @Mock private Twitter twitter;
+    @Mock private Status status;
+
+    @InjectMocks TwitterStatus twitterStatus;
+    @InjectMocks TwitterTimeline twitterTimeline;
+    @InjectMocks LiUniResource resource;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        twitterStatus.setTwitter(twitter);
+        resource.setTwitterStatus(twitterStatus);
+        twitterTimeline.setTwitter(twitter);
+        resource.setTwitterTimeline(twitterTimeline);
+    }
+
+    @After
+    public void tearDown() {
+        reset(twitter);
+        reset(status);
+    }
+
+    /* Test LiUniResource .fetchTimeline() REST method */
+    @Test
+    public void testREST_fetchTimeline_emptyTimeline() {
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        try {
+            doReturn(respList).when(twitter).getHomeTimeline();
+
+            Response resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.OK.getStatusCode());
+            assertEquals(resp.getEntity(), respList);
+        }
+        catch (Exception e) {
+            System.out.println("This exception is not expected.");
+            Assert.fail();
+        }
+    }
+
+    @Test
+    public void testREST_fetchTimeline_updateTimeline() {
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        try {
+            doReturn(respList).when(twitter).getHomeTimeline();
+
+            Response resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.OK.getStatusCode());
+            assertEquals(resp.getEntity(), respList);
+
+            respList.add(status);
+            resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.OK.getStatusCode());
+            assertEquals(resp.getEntity(), respList);
+        }
+        catch (Exception e) {
+            System.out.println("This exception is not expected.");
+            Assert.fail();
+        }
+    }
+
+    @Test
+    public void testREST_fetchTimeline_fetchTimelineError() {
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        respList.add(status);
+        String expectedError = (new ErrorModel()).getGeneralError();
+        try {
+            when(twitter.getHomeTimeline()).thenThrow(new TwitterException(""));
+
+            Response resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            assertEquals(resp.getEntity().toString(), expectedError);
+        }
+        catch (Exception e) {
+            System.out.println("This exception is not expected.");
+            Assert.fail();
+        }
+    }
+
+    /* Test LiUniResource .postTweet() REST method */
+    @Test
+    public void testREST_postTweet_successfulPost() {
```

Comment:
another way to handle the unexpected exceptions is to "throw Exception" on the method level.  No need to change anything but more of an FYI